database/init.sql

```diff
@@ -1,7 +1,21 @@
 CREATE TABLE account (
   id SERIAL PRIMARY KEY,
   email TEXT UNIQUE NOT NULL,
-  password TEXT NOT NULL
+  is_enabled boolean NOT NULL DEFAULT false
+);
+
+create table account_confirmation
```

Comment:
Is there a reason for being too explicit with the definitions? often overriding with the default values as well?

Reply:
there is no any specific reason, its already assigning false from code, but for any case I put default

Reply:
I meant for the entire table in general. I am not quite sure how the comment ended down there. Anyway, if I am not mistaken, for a field like id you are assigning the default sequence explicitly for example (maybe not the max value?). Also, the associated behavior with the foreign key constraint is the default one as well.

Reply:
I created account_confirmation table to save token there. It has id which is primary key of table and is autoincrement. It is `int` and 2147483647 is max Integer value, PostgresSQL added it when I was exporting SQL script. Also has foreign key to have relationship with account which needs to be confirmed. There is one to one relationship in `AccountConfirmation` entity class. First we create account, then confirmation token, and assign already created account to it, and then save it in this table. When account is confirmed, column `is_enabled` in account becomes true and associated token will be deleted from account_confirmation table.

Reply:
The workflow is fine but since a disabled account still exists in the accounts table, we are typically going to have to do additional work when dealing with the table, however, maybe we do want disabled(non verified) accounts to be able to perform limited actions like simply logging in and editing his/her profile but on the other hand, we probably don't want him/her to be able to add devices. We should get the others' opinion on this one.

@chaoran-chen @dajenet @bieniusa what do you think?

Reply:
I think it is quite common that a not activated account cannot do anything, not even to log in, so that's fine for me.